MapboxNavigation/RouteVoiceController.swift

```diff
@@ -159,14 +161,24 @@ open class RouteVoiceController: NSObject, AVSpeechSynthesizerDelegate {
     }
     
     func duckAudio() throws {
-        let categoryOptions: AVAudioSessionCategoryOptions = [.duckOthers, .interruptSpokenAudioAndMixWithOthers]
-        try AVAudioSession.sharedInstance().setMode(AVAudioSessionModeSpokenAudio)
-        try AVAudioSession.sharedInstance().setCategory(AVAudioSessionCategoryPlayback, with: categoryOptions)
+        if #available(iOS 12.0, *) {
+            let categoryMode = CarPlayManager.isConnected ? AVAudioSessionModeVoicePrompt : AVAudioSessionModeSpokenAudio
+            try AVAudioSession.sharedInstance().setCategory(AVAudioSessionCategoryPlayback, mode: categoryMode, options: categoryOptions)
+        } else if #available(iOS 10.0, *) {
+            try AVAudioSession.sharedInstance().setCategory(AVAudioSessionCategoryPlayback, mode: AVAudioSessionModeSpokenAudio, options: categoryOptions)
```

Comment:
`setCategory(_:mode:options:)` is convenient, but it’s OK to set the mode and category separately as long as you call both setters before activating the audio session.

Reply:
Well, it certainly smashes the LOC in this context. Done ✅